pkg/handler/handler.go

```diff
@@ -96,26 +100,82 @@ type ReadmeUtgaveFilter struct {
 	First  *int32
 }
 
-func (r *resolver) ReadmeUtgaver(filter *ReadmeUtgaveFilter) ([]ReadmeUtgave, error) {
-	if filter == nil {
-		return r.readmes, nil
-	}
-	var filteredReadmes []ReadmeUtgave
-	for _, r := range r.readmes {
-		if filter.Year != nil && *filter.Year != r.Year {
-			continue
+func getLink(name string) string {
+	return fmt.Sprintf("https://storage.googleapis.com/%v/%v", bucketName, name)
+}
+
+func getReadmes(ctx context.Context, client storage.Client, query string) ([]ReadmeUtgave, error) {
+	pdfQuery := &storage.Query{Prefix: "pdf/" + query}
+	imageQuery := &storage.Query{Prefix: "images/" + query}
+
+	re := regexp.MustCompile(`(?P<Year>\d{4})-(?P<Utgave>\d{2})`)
+
+	bkt := client.Bucket(bucketName)
+	pdfIt := bkt.Objects(ctx, pdfQuery)
+	imageIt := bkt.Objects(ctx, imageQuery)
+	var err error
+	var readmes  []ReadmeUtgave
+	for {
+		pdfAttrs, err := pdfIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		imageAttrs, err := imageIt.Next()
+
+		if err == iterator.Done {
+			break
 		}
-		if filter.Utgave != nil && *filter.Utgave != r.Utgave {
+
+		// Object may be a directory
+		match := re.FindAllString(pdfAttrs.Name, -1)
+		if len(match) == 0 {
 			continue
 		}
-		filteredReadmes = append(filteredReadmes, r)
-		if filter.First != nil && len(filteredReadmes) == int(*filter.First) {
-			break
+
+		matches := re.FindStringSubmatch(pdfAttrs.Name)
+		utgave, _ := strconv.ParseInt(matches[2], 10, 32)
+		year, _ := strconv.ParseInt(matches[1], 10, 32)
+		title := re.FindAllString(pdfAttrs.Name, 1)[0]
+
+		r := ReadmeUtgave{
+			Title: title,
+			Image: getLink(imageAttrs.Name),
+			Pdf: getLink(pdfAttrs.Name),
+			Year: int32(year),
+			Utgave: int32(utgave),
 		}
+		readmes = append(readmes, r)
+	}
+	if err == iterator.Done {
```

Comment:
Maybe remove this, since it will never be true?

Reply:
It will.https://github.com/webkom/readme-as-a-function/blob/7d5bb33b75e47e501bfd443eb4e624963b627e2f/pkg/handler/handler.go#L120-L122
When iterating, we break if the iterator is done, thus `err = iterator.Done`. Maybe not the prettiest, but this check ensures that the error is "correct".

Reply:
The err inside the for loop is scoped inside the loop, so this err will always be nil

Reply:
lol i'm an idiot